metadata-ingestion/src/datahub/sdk/mlmodelgroup.py

```diff
@@ -0,0 +1,217 @@
+        self._ensure_model_group_props().name = display_name
+
+    @property
+    def description(self) -> Optional[str]:
+        return self._ensure_model_group_props().description
+
+    def set_description(self, description: str) -> None:
+        self._ensure_model_group_props().description = description
+
+    @property
+    def external_url(self) -> Optional[str]:
+        return self._ensure_model_group_props().externalUrl
+
+    def set_external_url(self, external_url: str) -> None:
+        self._ensure_model_group_props().externalUrl = external_url
+
+    @property
+    def custom_properties(self) -> Optional[Dict[str, str]]:
+        return self._ensure_model_group_props().customProperties
+
+    def set_custom_properties(self, custom_properties: Dict[str, str]) -> None:
+        self._ensure_model_group_props().customProperties = custom_properties
+
+    @property
+    def created(self) -> Optional[datetime]:
+        return parse_time_stamp(self._ensure_model_group_props().created)
+
+    def set_created(self, created: datetime) -> None:
+        self._ensure_model_group_props().created = make_time_stamp(created)
+
+    @property
+    def last_modified(self) -> Optional[datetime]:
+        return parse_time_stamp(self._ensure_model_group_props().lastModified)
+
+    def set_last_modified(self, last_modified: datetime) -> None:
+        self._ensure_model_group_props().lastModified = make_time_stamp(last_modified)
+
+    @property
+    def training_jobs(self) -> Optional[List[str]]:
+        return self._ensure_model_group_props().trainingJobs
+
+    def set_training_jobs(self, training_jobs: List[str]) -> None:
+        self._ensure_model_group_props().trainingJobs = training_jobs
+
+    def add_training_jobs(self, training_jobs: List[str]) -> None:
+        props = self._ensure_model_group_props()
+        if props.trainingJobs is None:
+            props.trainingJobs = []
+        props.trainingJobs.extend(training_jobs)
+
+    def remove_training_jobs(self, training_jobs: List[str]) -> None:
+        props = self._ensure_model_group_props()
+        if props.trainingJobs is not None:
+            props.trainingJobs = [
+                job for job in props.trainingJobs if job not in training_jobs
+            ]
+
+    @property
+    def downstream_jobs(self) -> Optional[List[str]]:
+        return self._ensure_model_group_props().downstreamJobs
+
+    def set_downstream_jobs(self, downstream_jobs: List[str]) -> None:
+        self._ensure_model_group_props().downstreamJobs = downstream_jobs
+
+    def add_downstream_jobs(self, downstream_jobs: List[str]) -> None:
+        props = self._ensure_model_group_props()
+        if props.downstreamJobs is None:
+            props.downstreamJobs = []
+        props.downstreamJobs.extend(downstream_jobs)
+
+    def remove_downstream_jobs(self, downstream_jobs: List[str]) -> None:
+        props = self._ensure_model_group_props()
+        if props.downstreamJobs is not None:
```

Comment:
why do we need this if condition? the type annotation makes it look like this will always be true?